rcl/test/rcl/test_info_by_topic.cpp

```diff
@@ -0,0 +1,370 @@
+      false,
+      &this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_null_allocator)
+{
+  const auto ret = rcl_get_subscriptions_info_by_topic(&this->node, nullptr, this->topic_name,
+      false,
+      &this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_info_by_topic_null_topic)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  const auto ret = rcl_get_publishers_info_by_topic(&this->node,
+      &allocator, nullptr, false, &this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_null_topic)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  const auto ret = rcl_get_subscriptions_info_by_topic(&this->node,
+      &allocator, nullptr, false, &this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_info_by_topic_null_participants)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  const auto ret = rcl_get_publishers_info_by_topic(&this->node,
+      &allocator, this->topic_name, false, nullptr);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_null_participants)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  const auto ret = rcl_get_subscriptions_info_by_topic(&this->node,
+      &allocator, this->topic_name, false, nullptr);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestInfoByTopicFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_info_by_topic_invalid_participants)
+{
+  // temp_info_array is invalid because it is expected to be zero initialized
+  // and the info_array variable inside it is expected to be null.
+  const auto & temp_info_array = &this->topic_info_array;
+  temp_info_array->info_array =
+    static_cast<rmw_topic_info_t *>(calloc(1, sizeof(rmw_topic_info_t)));
```

Comment:
@mm318 I think `new rwm_topic_info_t()` may achieve what you want (see Construction section on [`new` expression documentation](https://en.cppreference.com/w/cpp/language/new)).

Reply:
Oops, forgot about C++03 features :)

I have changed the use of `calloc()` to using `new()`.